scipy/stats/_distn_infrastructure.py

```diff
@@ -1283,21 +1291,37 @@ def moment(self, n, *args, **kwds):
                 mdict = {'moments': {1: 'm', 2: 'v', 3: 'vs', 4: 'vk'}[n]}
             else:
                 mdict = {}
-            mu, mu2, g1, g2 = self._stats(*args, **mdict)
-        val = _moment_from_stats(n, mu, mu2, g1, g2, self._munp, args)
+            mu, mu2, g1, g2 = self._stats(*shapes, **mdict)
+        val = np.empty(loc.shape)  # val needs to be indexed by loc
+        val[...] = _moment_from_stats(n, mu, mu2, g1, g2, self._munp, shapes)
 
         # Convert to transformed  X = L + S*Y
         # E[X^n] = E[(L+S*Y)^n] = L^n sum(comb(n, k)*(S/L)^k E[Y^k], k=0...n)
-        if loc == 0:
-            return scale**n * val
-        else:
-            result = 0
-            fac = float(scale) / float(loc)
+        result = zeros(i0.shape)
+        place(result, ~i0, self.badvalue)
+
+        if i1.any():
+            res1 = scale[loc == 0]**n * val[loc == 0]
+            place(result, i1, res1)
+
+        if i2.any():
+            if mu is not None:
```

Comment:
I'm concerned about this line. There are some distributions for which `mu` might not be none but for which the others (`mu2`, etc...) will be `None`.

Reply:
Looks like the test failure might be caused by that. I thought this was ready, but I guess not : /

Reply:
> Looks like the test failure might be related because `dweibull` is one of those distributions. I thought this was ready, but I guess not : /

I have tried to resolve this in the latest commit. Do you think the changes are OK? If yes, is it still possible to get this merged for 1.7?

Reply:
Since it is labelled a bug fix it may need to get backported before the rc or during the RC phase, which is fine if it really is important enough to do that.

Reply:
But probably not before I branch, since I'm about to do that.

Reply:
Thanks @tylerjereddy for the replies! I wouldn't say it's top priority but it would be nice to have it in 1.7 since it has been up for a while now and fixes quite a bit to avoid workarounds in other PRs. Though, if it means a lot of work has to be done from your side, it would be fine to merge this later too.

Reply:
Shouldn't be a problem to put it in, I just mean it may get delayed until after I branch and then get added in with other bug fixes shortly after branching.

Reply:
> Shouldn't be a problem to put it in, I just mean it may get delayed until after I branch and then get added in with other bug fixes shortly after branching.

Oh, OK. That's fine! Thanks for the clarification.